archive/brave-web-servlet-filter/src/main/java/com/github/kristofa/brave/util/ModifiedUrlEncoder.java

```diff
@@ -0,0 +1,318 @@
+/**
+ * Modified java.net.UrlEncoder to process gracefull convercion of URLs
+ */
+
+/*
+ * Copyright (c) 1995, 2013, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
```

Comment:
sadly we can't do this as it is GPL. appreciate the thought, though.

how about looking at okhttp's HttpUrl type? you can either copy/paste from it or shade it I think

Reply:
:D Ok, I'll look into it. Became too personal to let it go)